src/mbgl/map/map.cpp

```diff
@@ -899,10 +899,15 @@ std::unique_ptr<Layer> Map::removeLayer(const std::string& id) {
     impl->styleMutated = true;
     BackendScope guard(impl->backend);
 
-    auto removedLayer = impl->style->removeLayer(id);
-    impl->onUpdate(Update::Classes);
-
-    return removedLayer;
+    try {
+        auto removedLayer = impl->style->removeLayer(id);
+        impl->onUpdate(Update::Classes);
+        return removedLayer;
+    } catch (const std::runtime_error& err) {
+        impl->styleMutated = false;
```

Comment:
Should I remove this statement? Since a couple of lines above there's `impl->styleMutated = true;` assuming the layer will be removed. And since this is in a `catch` block - the layer is not removed and the style is not mutated.